src/smc-webapp/buttonbar.coffee

```diff
@@ -51,11 +51,280 @@ exports.commands =
+
+        connect_to_a_remote_repository :
+                insert:
+                        """
+                        git remote add origin <server>
+                        """
+
+        list_all_currently_configured_remote_repositories :
+                insert:
+                        """
+                        git remote -v
+                        """
+
+        branches_create_a_new_branch_and_switch_to_it :
+                insert:
+                        """
+                        git checkout -b <branchname>
+                        """
+
+        switch_from_one_branch_to_another :
+                insert:
+                        """
+                        git checkout <branchname>
+                        """
+
+        list_all_the_branches_in_your_repo_and_also_tell_you_what_branch_youre_currently_in :
+                insert:
+                        """
+                        git branch
+                        """
+
+        delete_the_feature_branch :
+                insert:
+                        """
+                        git branch -d <branchname>
+                        """
+
+        push_the_branch_to_your_remote_repository_so_others_can_use_it :
+                insert:
+                        """
+                        git push origin <branchname>
+                        """
 
+        push_all_branches_to_your_remote_repository :
+                insert:
+                        """
+                        git push --all origin
+                        """
+
+        delete_a_branch_on_your_remote_repository :
+                insert:
+                        """
+                        git push origin :<branchname>
+                        """
+
+        update_from_the_remote_repository_fetch_and_merge_changes_on_the_remote_server_to_your_working_directory :
+                insert:
+                        """
+                        git pull
+                        """
+
+        to_merge_a_different_branch_into_your_active_branch :
+                insert:
+                        """
+                        git merge <branchname>
+                        """
+
+        view_all_the_merge_conflicts :
+                insert:
+                        """
+                        git diff
+                        """
+
+        git_diff_base_filename :
+                insert:
+                        """
+                        git diff --base <filename>
+                        """
+
+        git_diff_sourcebranch_targetbranch :
+                insert:
+                        """
+                        git diff <sourcebranch> <targetbranch>
+                        """
+
+        after_you_have_manually_resolved_any_conflicts_you_mark_the_changed_file :
+                insert:
+                        """
+                        git add <filename>
+                        git tag 1.0.0 <commitID>
+                        git log
+                        """
```

Comment:
I would remove this, I don't know if those three commands always make much sense. 

Instead, I would add `git log` alone, because there is no mention of being able to show the history of previous commits!